### DISCUSSION TITLE
Clickable Marker v7

### DISCUSSION THREAD

**Discussion by dreamorosi** [uber/react-map-gl#1744] [Q&A] [answered]
Is it possible on `v7.x` to make a `Marker` component rendered on a map clickable?

Based on the documentation there's no `onClick` callback anymore but only `onDrag`, `onDragStart`, `onDragEnd`. I see that there's a property called `clickTolerance` that suggests the possibility of distinguishing between a click & a drag.

I've tried with different combinations of `draggable={true}`, `onDrag*`, and `clickTolerance` but I either make the marker draggable or not interactive at all.

My use case is quite trivial, I just would like to render `Popup` component when the `Marker` one is clicked but I would like said marker to stay where it is in all circumstances.

EDIT: I've tried to also add `style={{ pointerEvents: "all", }}` on the `Marker` like suggested in this other discussion but that didn't help.

The only way that I can make a `Maker` clickable is if I render a child inside it and make the child clickable.

So I guess the new question is: is it not possible to make the default marker clickable? The use case would be to want to use the default marker & make it clickable. With the child method I need to bring my own icon & all.

**Comment by Pessimistress** [comment, marked as answer]
https://visgl.github.io/react-map-gl/examples/controls

**Reply by dreamorosi** [reply]
Yes, as I mentioned in my edit I figured out that adding the `onClick` on a child of `Marker` works. 
But this means I cannot make a default marker (i.e. the one rendered if no child is passed) clickable?

**Reply by dreamorosi** [reply]
To be clear, this:
```js
<Marker onClick={...} color="red" />
```
is not supported?

Only this is:
```js
<Marker ><MyOwnPinIcon onClick={...}/></Marker>
```
Is that correct?

**Reply by Pessimistress** [reply]
Correct, this use case was overlooked because mapbox-gl's `Marker` does not have a `click` event. Could you open a feature request issue? I can take a look some time this weekend.

**Reply by dreamorosi** [reply]
Feature request has been opened here, let me know if it's missing any info: https://github.com/visgl/react-map-gl/issues/1746
